### PR TITLE
Enable Active Memory Manager by default

### PR DESCRIPTION
**Pull request by crusaderky**
This PR
- enables the AMM ReduceReplicas policy globally
- changes client.retire_workers and adaptive downscaling from spinning up a temporary AMM instance to using the permanent one.
- explicitly disables AMM in 80 tests that perform assertions around replicas, call replicate() or scatter(broadcast=True), etc.

Evidence that the Active Memory Manager is universally desirable:
https://github.com/coiled/coiled-runtime/issues/140

### CODE REVIEW

**Comment by crusaderky** [comment]
Blocked by #7059

File: distributed/tests/test_steal.py
Comment (crusaderky) [inline distributed/tests/test_steal.py:1152]: @gjoseph92 this is why the test was flaky with queueing: there were key collisions. The problem was actually unrelated to queueing and just a matter of subtle timings; it became flaky with AMM too.

File: distributed/tests/test_scheduler.py
Comment (crusaderky) [inline distributed/tests/test_scheduler.py:1358]: This was causing a key collision

**Comment by crusaderky** [comment]
Ran 7 times with 0.01s interval; everything looks stable. Ready for final review and merge.

File: docs/source/resilience.rst
Comment (hendrikmakait) [inline docs/source/resilience.rst:56]:
```suggestion
    factor to ensure that it remains long-lived or else back the data off on some
```

File: docs/source/active_memory_manager.rst
Comment (hendrikmakait) [inline docs/source/active_memory_manager.rst:135]:
```suggestion
any task, this policy uninstalls itself automatically from the Active Memory Manager and
```

File: distributed/utils_test.py
Comment (hendrikmakait) [inline distributed/utils_test.py:115]: nit:
```suggestion
NO_AMM_CONFIG = {"distributed.scheduler.active-memory-manager.start": False}
```

File: distributed/utils_test.py
Comment (hendrikmakait) [inline distributed/utils_test.py:538]: This docstring is confusing me. Isn't this actively turning off AMM? Also, do we need the `assert not client.amm.running()` after `yield client`? Shouldn't we just ensure that AMM is reset to `before`?

**Review by hendrikmakait — COMMENTED** [review]
Overall, this looks good to me. I have a couple of nits and the `client_no_amm` is confusing to me, I'd appreciate some clarification on that.

File: distributed/utils_test.py
Comment (crusaderky) [reply distributed/utils_test.py:115]: I find it very redundant and verbose to see `config=NO_AMM_CONFIG` every time

Comment (hendrikmakait) [reply distributed/utils_test.py:115]: Maybe an alternative solution would be a TypeAlias to make it clearer that the `NO_AMM` constant is a configuration?

File: distributed/utils_test.py
Comment (crusaderky) [reply distributed/utils_test.py]: Revisited the fixture

File: distributed/utils_test.py
Comment (crusaderky) [reply distributed/utils_test.py:115]: Such a TypeAlias would need to go in dask/dask, and I wouldn't be able to enforce it in the signature of gen_cluster anyway.
Would it be OK if I just wrote an explanatory comment on the definition?

Comment (hendrikmakait) [reply distributed/utils_test.py:118]: Fair point, the TypeAlias doesn't warrant any hold-up on this PR. Thanks for adding a comment. Generally with nits, feel free to ignore them :)

**Review by hendrikmakait — APPROVED** [review]
LGTM assuming CI doesn't suddenly start failing after the latest changes. Thanks @crusaderky!

**Pull request merged**
